apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -123,8 +124,47 @@ create_contract(_Cfg) ->
     Tx           = aect_test_utils:create_tx(PubKey, Overrides, S1),
 
     %% Test that the create transaction is accepted
-    {ok, S2}   = sign_and_apply_transaction(Tx, PrivKey, S1, ?MINER_PUBKEY),
-    {{value, _}, _} = lookup_contract_by_id(aect_contracts:compute_contract_pubkey(PubKey, aetx:nonce(Tx)), S2),
+    {ok, S2} = sign_and_apply_transaction(Tx, PrivKey, S1, ?MINER_PUBKEY),
+    %% Check that the contract is created
+    ContractKey = aect_contracts:compute_contract_pubkey(PubKey, aetx:nonce(Tx)),
+    {{value, Contract}, _} = lookup_contract_by_id(ContractKey, S2),
+    %% Check that the created contract has the correct details from the contract create tx
+    ?assertEqual(PubKey, aect_contracts:owner(Contract)),
+    ?assertEqual(aect_create_tx:vm_version(aetx:tx(Tx)), aect_contracts:vm_version(Contract)),
+    ?assertEqual(aect_create_tx:code(aetx:tx(Tx)), aect_contracts:code(Contract)),
+    ?assertEqual(aect_create_tx:deposit(aetx:tx(Tx)), aect_contracts:deposit(Contract)),
+    %% Check that the created contract has the correct details not from the contract create tx
+    _ = aect_contracts:log(Contract),
+    ?assert(aect_contracts:active(Contract)),
+    ?assertEqual([], aect_contracts:referers(Contract)),
+    %% Check that the contract init call is created
+    ?assertEqual([], aect_call_state_tree:to_list(aect_test_utils:calls(S1))),
+    ?assertMatch([_], aect_call_state_tree:to_list(aect_test_utils:calls(S2))),
+    InitCallId = aect_call:id(PubKey, aetx:nonce(Tx), ContractKey),
+    {value, InitCall} = aect_call_state_tree:lookup_call(ContractKey, InitCallId, aect_test_utils:calls(S2)),
+    %% Check that the created init call has the correct details from the contract create tx
+    ?assertEqual(PubKey, aect_call:caller_address(InitCall)),
+    ?assertEqual(aetx:nonce(Tx), aect_call:caller_nonce(InitCall)),
+    %% Check that the created init call has the correct details not from the contract create tx
+    ?assertEqual(ContractKey, aect_call:contract_address(InitCall)),
+    _ = aect_call:height(InitCall), %% Unclear if this needed.
+    _ = aect_call:gas_used(InitCall), %% No init in contract - shall value be zero?
```

Comment:
Shall value be zero? (I checked and it is positive.)

Reply:
@happi Does this relate to ["Make sure init is executed at contract create"](https://www.pivotaltracker.com/story/show/158109766) ?

Reply:
Yes. In the Sophia ABI an init function is required in the contract and create must call the init function.

Reply:
If no init is defined the compiler inserts a stub, so the gas cost will be positive, but small.